pkg/api/steve/proxy/proxy.go

```diff
@@ -145,12 +146,15 @@ func (h *Handler) authLocalCluster(router *gmux.Router) func(rw http.ResponseWri
 
 func (h *Handler) matchManagementCRDs() gmux.MatcherFunc {
 	return func(req *http.Request, match *gmux.RouteMatch) bool {
-		splitPath := strings.Split(req.URL.Path, "/")
-		if len(splitPath) < 3 {
+		gvr, err := gvrFromPath(req.URL.Path)
+		if err != nil {
 			return false
 		}
-		apiGroup := splitPath[2]
-		return apiGroup == managementv3.GroupName
+		// Only match v3 management CRDs that are not in the allow-list.
+		return gvr.Group == managementv3.GroupName &&
+			gvr.Version == managementv3.Version &&
+			gvr.Resource != "" &&
```

Comment:
I don't think that this maintains existing behavior. For example, if we have an unrecognized path that is recognizable for the management api group such as:

> `apis/management.cattle.io/v3/namespaces/default/fake-crd/some/other/stuff/that/makes/this/long`

Today we will block access to that endpoint. But in this model we will let that through - since the length is not recognized, resource will be "" and we won't match. I think that default on is a big dangerous for this function, and would recommend that we change it.